김현수/mvvmstudy/app/src/main/java/com/example/mvvmstudy/UI/MainActivity.kt

```diff
@@ -0,0 +1,176 @@
+    private var search =""
+
+    private var _binding : ActivityMainBinding?= null
+    private val binding get() = _binding!!
+    private val adapter = RepositoryAdapter(this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        binding.listRepo.adapter = adapter
+        binding.listRepo.layoutManager = LinearLayoutManager(this)
+
+        initScrollListener()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.main_menu,menu)
+        val searchItem = menu?.findItem(R.id.item_search)
+        val searchView = searchItem?.actionView as SearchView
+        searchView.queryHint = "Search.."
+
+        searchView.setOnQueryTextListener(object:SearchView.OnQueryTextListener{
+            override fun onQueryTextSubmit(p: String?): Boolean {
+                if(search != p.toString()) page = 0
+               fetching(p.toString())
+                search = p.toString()
+
+                // 요녀석 땜에 꽤 고생
+                searchView.clearFocus()
+               return false
+            }
+            override fun onQueryTextChange(p0: String?): Boolean {
+                return false
+            }
+        })
+
+        return super.onCreateOptionsMenu(menu)
+    }
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when(item?.itemId){
+            android.R.id.home ->{
+                finish()
+                return true
+            }
+            else ->{
+                return super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+
+    fun fetching(q: String){
+        RetrofitClass.retrofitService
+            .searchRepositories(q,getPage(),50)
+            .enqueue(object: Callback<Repositories>{
+                val progress = indeterminateProgressDialog(message = "Loading...")
+                @SuppressLint("NotifyDataSetChanged")
+                override fun onResponse(
+                    call: Call<Repositories>,
+                    response: Response<Repositories>
+                ) {
+                    progress.show()
+                    when(response!!.code()){
+                        200->{
+                            val body = response.body() as Repositories
+                            adapter.setRepository(body.repositories)
+                            Log.d("api","Ok")
+                            progress.cancel()
+                        }
+                        304->{
+                            Log.d("api","Not modified")
+                            progress.cancel()
+                        }
+                        422 ->{
+                            Log.d("api","Validation failed")
+                            progress.cancel()
+                        }
+                        503->{
```

Comment:
요기도 codeformat 하면 좋을 것 같아요!
```suggestion
                        200 -> {
                            val body = response.body() as Repositories
                            adapter.setRepository(body.repositories)
                            Log.d("api","Ok")
                            progress.cancel()
                        }
                        304 -> {
                            Log.d("api","Not modified")
                            progress.cancel()
                        }
                        422 -> {
                            Log.d("api","Validation failed")
                            progress.cancel()
                        }
                        503 -> {
```

Reply:
code format.. 끄적끄적.. 난리가 났었군요 ㅎㅎ..